app/code/Magento/PageCache/Model/Varnish/VclGenerator.php

```diff
@@ -228,4 +239,18 @@ private function getDesignExceptions()
     {
         return $this->designExceptions;
     }
+
+    /**
+     * Check if document root contains pub
+     *
+     * @return string
+     */
+    private function getHealthCheck()
+    {
+        if (strpos($this->request->getServer('DOCUMENT_ROOT'), 'pub') === false) {
```

Comment:
I'd base the check on the configuration rather than on the current request.

What do you think about using `\Magento\Framework\Url::getBaseUrl` ?

Reply:
Maybe I got it wrong, but doesn't `\Magento\Framework\Url::getBaseUrl` return the configured Base URL? So it doesn't provide any information about the docroot. For this function we have to check if the configured directory is e.g. `/var/www/html` or `/var/www/html/pub`.

Reply:
Configured base URL would or would not include "pub" directory depending on the configuration. `getBaseUrl` seems safer for me than `DOCUMENT_ROOT` as probably the webserver setup can be much more versatile than Magento configuration..

Reply:
But according to [DevDocs](https://devdocs.magento.com/guides/v2.3/install-gde/tutorials/change-docroot-to-pub.html#1-edit-your-server-configuration) the "pub" directory is configured via the server configuration. So do we have to change something over there, too?

Reply:
That's correct, I don't think anything there should be changed.

I'm just trying to be cautious about the case when Magento admin panel and frontend are on different virtual hosts or different servers with document roots configured in a different way. In this case, the Magento configuration is the better place to get the actual information about the way frontend is setup.

Reply:
I changed it now.